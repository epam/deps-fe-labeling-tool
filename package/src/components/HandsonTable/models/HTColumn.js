
import PropTypes from 'prop-types'

class HTColumn {
  constructor (data) {
    this.data = data
  }
}

const htColumnShape = PropTypes.shape({
  data: PropTypes.string.isRequired
})

export {
  HTColumn,
  htColumnShape
}
