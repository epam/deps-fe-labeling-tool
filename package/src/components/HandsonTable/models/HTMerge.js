import PropTypes from 'prop-types'

class HTMerge {
  constructor (row, col, rowspan, colspan) {
    this.row = row
    this.col = col
    this.rowspan = rowspan
    this.colspan = colspan
  }
}

const htMergeShape = PropTypes.shape({
  row: PropTypes.number.isRequired,
  col: PropTypes.number.isRequired,
  rowspan: PropTypes.number.isRequired,
  colspan: PropTypes.number.isRequired
})

export {
  HTMerge,
  htMergeShape
}
