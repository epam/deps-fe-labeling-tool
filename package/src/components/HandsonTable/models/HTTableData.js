
import PropTypes from 'prop-types'

const rowOfPrimitivesShape = PropTypes.arrayOf(
  PropTypes.oneOfType([
    PropTypes.number.isRequired,
    PropTypes.string.isRequired
  ])
)

const rowsOfPrimitivesShape = PropTypes.arrayOf(
  rowOfPrimitivesShape
)

export { rowsOfPrimitivesShape }
