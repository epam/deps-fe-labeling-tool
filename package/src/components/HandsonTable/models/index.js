
export * from './HTTableData'
export * from './HTMerge'
export * from './HTColumn'
