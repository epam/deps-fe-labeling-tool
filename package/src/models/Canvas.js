import PropTypes from 'prop-types'

export const canvasShape = PropTypes.shape({
  add: PropTypes.func,
  backgroundColor: PropTypes.string,
  defaultCursor: PropTypes.string,
  discardActiveObject: PropTypes.func,
  findTarget: PropTypes.func,
  fire: PropTypes.func,
  fxRemove: PropTypes.func,
  getActiveObject: PropTypes.func,
  getActiveObjects: PropTypes.func,
  getCenter: PropTypes.func,
  getHeight: PropTypes.func,
  getObjects: PropTypes.func,
  getPointer: PropTypes.func,
  getWidth: PropTypes.func,
  getZoom: PropTypes.func,
  height: PropTypes.number,
  off: PropTypes.func,
  on: PropTypes.func,
  remove: PropTypes.func,
  renderAll: PropTypes.func,
  requestRenderAll: PropTypes.func,
  selection: PropTypes.bool,
  sendToBack: PropTypes.func,
  setActiveObject: PropTypes.func,
  setBackgroundColor: PropTypes.func,
  setBackgroundImage: PropTypes.func,
  setCursor: PropTypes.func,
  setHeight: PropTypes.func,
  setViewportTransform: PropTypes.func,
  setWidth: PropTypes.func,
  setZoom: PropTypes.func,
  viewportTransform: PropTypes.arrayOf(PropTypes.number),
  width: PropTypes.number,
  zoom: PropTypes.number,
  zoomToPoint: PropTypes.func
})
