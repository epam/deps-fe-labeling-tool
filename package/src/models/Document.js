import PropTypes from 'prop-types'
import { KnownLanguage } from '@/enums/KnownLanguage'

class Document {
  constructor (pages, language = null, name = '', engine = '', extraName = '') {
    this.pages = pages
    this.language = language
    this.name = name
    this.engine = engine
    this.extraName = extraName
  }
}

const documentShape = PropTypes.shape({
  language: PropTypes.oneOf(
    Object.values(KnownLanguage)
  ),
  pages: PropTypes.arrayOf(PropTypes.string).isRequired,
  name: PropTypes.string.isRequired,
  engine: PropTypes.string.isRequired,
  extraName: PropTypes.string
})

export {
  documentShape,
  Document
}
