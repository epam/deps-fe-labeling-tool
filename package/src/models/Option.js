import PropTypes from 'prop-types'

class Option {
  constructor (value, text, disabled) {
    this.value = value
    this.text = text || value
    this.disabled = disabled
  }
}

const optionShape = PropTypes.shape({
  text: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  disabled: PropTypes.bool
})

export {
  Option,
  optionShape
}
