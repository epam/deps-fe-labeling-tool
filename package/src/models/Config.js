import PropTypes from 'prop-types'
import { documentShape } from '@/models/Document'
import { fieldShape } from '@/models/Field'
import { markupShape } from '@/models/Markup'
import { ocrEngineShape } from '@/models/OcrEngine'
import { ocrLanguageShape } from '@/models/OcrLanguage'
import { settingsShape } from '@/models/Settings'

class Config {
  constructor (document, fields, ocr, api, markup, settings, events) {
    this.document = document
    this.fields = fields
    this.ocr = ocr
    this.api = api
    this.markup = markup
    this.settings = settings
    this.events = events
  }
}

const ocrShape = PropTypes.shape({
  engines: PropTypes.arrayOf(ocrEngineShape),
  languages: PropTypes.arrayOf(ocrLanguageShape)
})

const apiShape = PropTypes.shape({
  close: PropTypes.func.isRequired,
  getImage: PropTypes.func,
  save: PropTypes.func.isRequired,
  saveMarkup: PropTypes.func,
  ocrTable: PropTypes.func,
  ocrText: PropTypes.func,
  addFieldForm: PropTypes.func
})

const eventsShape = PropTypes.shape({
  onClose: PropTypes.func
})

const configShape = PropTypes.shape({
  document: documentShape.isRequired,
  fields: PropTypes.arrayOf(fieldShape).isRequired,
  ocr: ocrShape,
  api: apiShape.isRequired,
  markup: markupShape,
  settings: settingsShape,
  events: eventsShape
})

export {
  apiShape,
  configShape,
  Config
}
