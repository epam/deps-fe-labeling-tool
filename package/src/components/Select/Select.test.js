import React from 'react'
import { shallow } from 'enzyme'
import { Option } from '@/models/Option'
import {
  Select,
  SelectMode,
  enumToOptions,
  keyValueToOptions
} from './Select'

const MOCK_VALUE_1 = 'MOCK_VALUE_1'
const MOCK_VALUE_2 = 'MOCK_VALUE_2'

describe('Component: Select', () => {
  let defaultProps
  let wrapper

  beforeEach(() => {
    defaultProps = {
      allowSearch: true,
      allowClear: true,
      placeholder: 'MOCK_PLACEHOLDER',
      value: [MOCK_VALUE_1],
      onChange: jest.fn(),
      options: [
        new Option(MOCK_VALUE_1, 'MOCK_TEXT_1'),
        new Option(MOCK_VALUE_2, 'MOCK_TEXT_2')
      ],
      dropdownRender: jest.fn(),
      onSearch: jest.fn(),
      dropdownMatchSelectWidth: true,
      mode: SelectMode.MULTIPLE
    }

    wrapper = shallow(<Select {...defaultProps} />)
  })

  it('should render antd Select with correct value and options', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass correct onChange callback to antd Select', () => {
    expect(wrapper.props().onChange).toEqual(defaultProps.onChange)
  })

  it('should pass correct filterOption callback to the antd Select', () => {
    expect(wrapper.props().filterOption).toEqual(wrapper.instance().filterOption)
  })

  it('should filter options in case insensitive manner', () => {
    const options = [{
      content: 'Minsk',
      expected: true
    }, {
      content: 'Paris',
      expected: false
    }, {
      content: 'Milan',
      expected: true
    }, {
      content: {},
      expected: false
    }]

    const searchStr = 'mI'

    options.forEach((o) => {
      const option = {
        props: {
          children: o.content
        }
      }

      const output = wrapper.instance().filterOption(searchStr, option)
      expect(output).toEqual(o.expected)
    })
  })

  it('should getContainer be passed in case of the absence of the getPopupContainer prop', () => {
    expect(wrapper.props().getPopupContainer).toEqual(wrapper.instance().getContainer)
  })

  it('should render with props.getPopupContainer if it was passed', () => {
    const getPopupContainer = jest.fn()
    wrapper.setProps({ ...defaultProps, getPopupContainer })
    expect(wrapper.props().getPopupContainer).toEqual(getPopupContainer)
  })

  it('should keyValueToOptions return array of Option', () => {
    const arrOptions = keyValueToOptions({ test: 'test1', test2: 'test3' })
    expect(arrOptions).toEqual([new Option('test', 'test1'), new Option('test2', 'test3')])
  })

  it('should enumToOptions return array of Option', () => {
    const arrOptions = enumToOptions({ test: 'test', test1: 'test1' })
    expect(arrOptions).toEqual([new Option('test'), new Option('test1')])
  })
})
