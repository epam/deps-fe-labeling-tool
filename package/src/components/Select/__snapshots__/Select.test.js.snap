// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Select should render antd Select with correct value and options 1`] = `
<Selectstyles__Select
  allowClear={true}
  dropdownMatchSelectWidth={true}
  dropdownRender={[MockFunction]}
  filterOption={[Function]}
  getPopupContainer={[Function]}
  mode="multiple"
  onChange={[MockFunction]}
  onSearch={[MockFunction]}
  placeholder="MOCK_PLACEHOLDER"
  showSearch={true}
  value={
    Array [
      "MOCK_VALUE_1",
    ]
  }
>
  <Option
    key="MOCK_VALUE_1"
    value="MOCK_VALUE_1"
  >
    MOCK_TEXT_1
  </Option>
  <Option
    key="MOCK_VALUE_2"
    value="MOCK_VALUE_2"
  >
    MOCK_TEXT_2
  </Option>
</Selectstyles__Select>
`;
